File: cpp/se/ArDomainUndeleteCommand.cpp
#include "se/ArDomainUndeleteCommand.hpp"

#include "se/ArExtension.hpp"
#include "se/ArDomainObjectType.hpp"
#include "se/CommandType.hpp"
#include "xml/XMLHelper.hpp"
#include "xml/XStr.hpp"

namespace {
    class ArUndeleteCommandType : public CommandType
    {
        public:
            ArUndeleteCommandType() : CommandType (getCommandName()) {}
            std::string getCommandName() const { return "undelete"; }
            std::string toString() const { return "undelete"; }
    };    

    Extension& arExtension() {
        static Extension* arExt = new ArExtension();
        return *arExt;
    }

    const ArUndeleteCommandType undeleteCmdType;
    const ArDomainObjectType ardomType;
}; // anonymous namespace

ArDomainUndeleteCommand::ArDomainUndeleteCommand(
        const std::string &name) : ProtocolExtensionCommand(
            &undeleteCmdType, &ardomType, name, arExtension())
{
}

File: src/main/java/com/ausregistry/jtoolkit2/se/ArDomainUndeleteCommand.java
package com.ausregistry.jtoolkit2.se;

/**
 * Mapping of EPP urn:ar:params:xml:ns:arext-1.0 domainUndelete command
 * specified by the AusRegistry EPP extensions document. Use this class to
 * generate an AusRegistry-compliant XML document, given simple input
 * parameters. The toXML method in Command serialises this object to XML.
 */
public class ArDomainUndeleteCommand extends ProtocolExtensionCommand {
    private static final long serialVersionUID = 420815971854196895L;

    private static final CommandType UNDELETE_COMMAND_TYPE = new ArUndeleteCommandType();

    public ArDomainUndeleteCommand(String name) {
        super(UNDELETE_COMMAND_TYPE, ExtendedObjectType.AR_DOMAIN, name);
    }

    protected Extension getExtension() {
        return ExtensionImpl.AR;
    }
}

